src/toolkits/object_detection/object_detector.cpp

```diff
@@ -1069,8 +1075,11 @@ void object_detector::init_training(gl_sframe data,
                                          read_state<int>("random_seed"));
 
   // Bind the data to a data iterator.
-  training_data_iterator_ = create_iterator(
-      training_data_, /* expected class_labels */ {}, /* repeat */ true);
+  std::vector<std::string> class_labels =
+      read_state<std::vector<std::string>>("classes");
+  training_data_iterator_ =
```

Comment:
should it be flex_list? or it can implicitly convert to std::vector<std::string>?

Reply:
I have tried to get<flex_List> and then construct a vector<string> from it.
Either method works here,
so I pick a concise one.

Reply:
That's really weird. Probably some conversion. I cannot assign vec<str> to vec<flexible>

Reply:
`vec str` to `vec flexible_type`

Reply:
I guess I mean
```
flex_list flex_classes = read_state<flex_list>("classes");
std::vector<std::string> class_labels (flex_classes.begin(), flex_classes.end());
```

Reply:
Yes I think it does some conversion :P

Reply:
oh well, it must do.

I mean
```
vector<string> blah = {"1", "1"};
flex_list foo = blah; // will error out
```